msger_mosaic_bridge/src/mosaic_client.cpp

```diff
@@ -181,28 +182,21 @@ void MosaicClient::received_vehicle_status(const std::shared_ptr<const std::vect
             pose[0] = vehicle_pose["x"].GetDouble();
             pose[1] = vehicle_pose["y"].GetDouble();
             pose[2] = vehicle_pose["z"].GetDouble();
+
+            // Calculate twist based on pose and prev_pose
+            for (int i = 0; i < 3; ++i) {
+                twist[i] = (pose[i] - prev_pose[i]) / time_difference;
```

Comment:
This will give us a weird result at the first timestep, where we get our first pose update, and our prev_pose is still 0,0,0. We'll get a very large twist for that frame right?

Might be worth setting twist to 0 for that first timestep. I'd recommend adding a bool member variable to track if it's the first update or not, and on that first update timestep we set twist to 0,0,0. Similar recommendation if there's any other cases where we're calculating a derivative like this.